### ISSUE TITLE
Update notification remains stuck if internet connection gets interrupted

### ISSUE THREAD

**Issue by buttercookie42** [breezy-weather/breezy-weather#535]
### Steps to reproduce

1. Leave your device offline for longer than the background refresh period.
2. Connect to the internet.
3. BreezyWeather starts refreshing the weather data, the corresponding notification appears.
4. Disconnect from the internet while the refresh is still running.

### Expected behavior

The "Updating weather data" notification is cancelled and disappears again.

### Actual behavior

The "Updating weather data" notification gets stuck. It cannot be swiped away, and pressing the "Cancel" button doesn't do anything either. I need to force stop the app to get rid of the notification. (Maybe waiting for the next background refresh to complete successfully once I have a more stable internet connection would work, too, but I didn't try that one.)

### Weather source used

AccuWeather

### Breezy Weather version

4.6.9-beta

### Android version

Android 6.0.1

### Device

Moto G4 Play

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I read the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document and followed instructions if relevant.
- [X] I have updated the app to **[version 4.6.9 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance), unless my issue only happens with a specific language
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by buttercookie42** [comment]
I've seen it happen with a flaky Wifi connection (and no mobile data), or if I've quickly turned off mobile data again before the weather update completed.

I'm not sure what happened to the update job specifically, but I can see the Breezy Weather service *and process* still running if I look at Android's Running Services list, and as far as I know Motorola isn't (or at least wasn't a few years ago, when my phone came out) particularly egregious about going beyond Android's standard battery optimisations. Certainly I haven't observed any particular problems with Breezy Weather as long as I have a stable network connection.

Curiously enough, when I tried it this time I didn't even get the additional notification about the failed update, although at other times I've seen that one, too. (Probably it depends on what moment the network connection exactly gets interrupted?)

**Comment by papjul** [comment]
> I've seen it happen with a flaky Wifi connection (and no mobile data), or if I've quickly turned off mobile data again before the weather update completed.

I use Breezy Weather daily and the only time it failed for network reasons was when I was going through a tunnel, but that notification didn't remain stuck, it just showed a notification error like the manual turn off network test I did.

> Curiously enough, when I tried it this time I didn't even get the additional notification about the failed update, although at other times I've seen that one, too. (Probably it depends on what moment the network connection exactly gets interrupted?)

I had to add 10 locations to be able to turn off quickly enough, because it's a very lightweight process and server responses from weather sources are usually very fast as well. First time I tried, the process completed successfully and I had no notification. If you're technical enough, you can build a debug version with Android Studio that you can install in parallel. The debug version contains a "Force weather update" button in Debug settings that allows you to quickly test without needing to wait ≥ 30 minutes.

I don't know if related but Android 6.x is rather old, unsure if it could be an old version-specific issue.

Let me know if you find anything or if I should tag it as "device-specific" for now.

**Comment by buttercookie42** [comment]
Yeah, I suppose it might something specific to ancient Android versions or even just specifically Motorola… in principle I can try playing around with a debug version, I just have to see when I can find some time for that.

**Comment by papjul** [comment]
Are you still experiencing issues with latest v5.2.7?

**Comment by buttercookie42** [comment]
I've since swapped my phone because it was getting too long in the tooth in the end, so I can't really say. Certainly the issue is no longer relevant for me that way.

**Comment by papjul** [comment]
Thanks for your reply. If you experience a similar issue, please open a new issue
